dotcom-rendering/cypress/e2e/parallel-4/show-more-test.cy.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable no-undef */
+/* eslint-disable func-names */
+import { disableCMP } from '../../lib/disableCMP';
+import { setLocalBaseUrl } from '../../lib/setLocalBaseUrl.js';
+
+import showMoreRequestBody from '../../fixtures/manual/science-show-more-req.json';
+import scienceFrontJson from '../../fixtures/manual/science-front.json';
+
+/**
+ * This tests both the '/Cards' endpoint, and the client-side 'Show More' i
+ * interaction that fetches and displays extra cards for a container.
+ * It mocks the responses from Frontend in order to keep the contents stable.
+ * This is partly to allow for testing properties like the URLs of the rendered
+ * cards, but also partly with a view to adding visual regression testing in the
+ * near future.
```

Comment:
This is a change to how we run Cypress. We currently don't mock data in cypress, we read it from production. There are well known pros and cons to this but I think it's important that we either stay consistent or, if we choose to diverge, we do so in a way that makes it clear which tests are mocked and which are not. Maybe this could be pulled out into a separate PR? 🙏

Reply:
okay yep that makes sense. I added mock data here partly because I'm anticipating adding a Percy snapshot at some point down the line, but happy to discuss further how we want to approach live vs. mock here.

When you say pull into a separate PR, do you mean take the test file out of this PR and open a separate PR for it? Or use live data here and then open a new PR to discuss mocking?

Reply:
Using live data in our end to end test is nice because it gives us extra security. If the test works on live data then it gives us a higher level of confidence, the scope of what we're testing is larger.

I think in this case we would need a static front though, one that isn't going to change over time. This is a lot easier to achieve using articles but might be possible?

Reply:
> This is a change to how we run Cypress. We currently don't mock data in cypress, we read it from production.

Worth noting that while in cypress we do go to live articles, we do often spoof some API responses with cy.intercept (e.g https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/cypress/lib/mocks.js).

I think the tradeoff to consider here is that while it's relatively easy for us to pick an older 'quiet' article where the chances of it changing in the future is very low - it may be harder to do this with fronts, and relying on specific containers/cards/layouts existing may result in us having to spend a lot more time fixing broken tests.

The advantage of having live data of course being that fixtures can get out-of-date, and tests can end up testing that out-of-date data works, rather than if 'live' data works.

TL;DR: I don't mind too much, but we do need to recognise that testing on fronts could require a different approach than testing on articles!

Reply:
> I don't mind too much, but we do need to recognise that testing on fronts could require a different approach than testing on articles

This seems right to me. Bracketing visual regression testing for now, it seems to me that in order to check the rendering logic of `ShowMore.Importable` here we'd need to write some pretty complex logic into the test, so that we could derive the number of cards that should be shown, and the urls, indices, etc that they should have.

I agree that we miss some things by mocking with static data though. Is it worth considering adding a test with live data in _addition_ to the mocked one?